Support multiple parameters

craigtsmith opened the pull request:
I found myself repeating calls to this action over and over like so:

```yaml
    - name: aws-ssm-to-env
      uses: bomb-on/aws-ssm-to-env@master
      with:
        ssm_parameter: 'PARAM_1'
        prefix: 'PREFIX_'

    - name: aws-ssm-to-env
      uses: bomb-on/aws-ssm-to-env@master
      with:
        ssm_parameter: 'PARAM_2'
        prefix: 'PREFIX_'

    - name: aws-ssm-to-env
      uses: bomb-on/aws-ssm-to-env@master
      with:
        ssm_parameter: 'PARAM_3'
        prefix: 'PREFIX_'
```

So I tweaked the bash script to accept either a singular string or a comma separated list of strings as inputs, the result being:

```yaml
    - name: aws-ssm-to-env
      uses: craigtsmith/aws-ssm-to-env@master
      with:
        ssm_parameter_list: | 
          PARAM_1, 
          PARAM_2, 
          PARAM_3
        prefix: 'PREFIX_'
```

Maybe somebody will find it useful :) 

Thanks for the action! Any changes you want me to make let me know.

craigtsmith committed:
Update action.yml

craigtsmith committed:
Update entrypoint.sh

craigtsmith committed:
Update action.yml

craigtsmith committed:
Update entrypoint.sh

craigtsmith committed:
Update README.md

bomb-on commented:
Thanks for your contribution! I see your issue here, makes sense to be able to do it simpler like you suggested, will review the code when I catch some time, definitely this week!

bomb-on commented on README.md:
```diff
@@ -63,11 +64,38 @@ jobs:
 
 Example above will set environmental variable `AWS_SSM_MY_PARAMETER_NAME` with value from the AWS SSM parameter itself.
 
+#### List of parameter values
```
Change this into `#### Get multiple parameter values` with description:

`Use comma separated list of strings to fetch multiple parameter values at once:`

bomb-on commented on README.md:
```diff
@@ -36,14 +37,14 @@ Action expects 3 secrets to be set in GitHub's repository:
 ### Examples
 
-#### String values
+#### Singular parameter value
```
Change this into `#### Get single parameter value`

bomb-on commented on README.md:
```diff
@@ -36,14 +37,14 @@ Action expects 3 secrets to be set in GitHub's repository:
 Parameter name | Type | Required | Default Value | Description
 --- | --- | --- | --- | ---
-`ssm_parameter` | string | true | | AWS Systems Manager parameter name (path)
+`ssm_parameter_list` | string | true | | AWS Systems Manager parameter name (path) or comma seperated list of paths
```
"seperated" should be "separated"

bomb-on commented on README.md:
```diff
@@ -9,7 +9,8 @@
      * [Parameters](#parameters)
      * [Examples](#examples)
-        * [String values](#string-values)
+        * [Singular paramter value](#singular-parameter-value)
```
Change this according to changes below (also watch for typos - `paramter` ;))

bomb-on commented on README.md:
````diff
@@ -63,11 +64,38 @@ jobs:
+           my_second_parameter
+```
+
+Example above will set environmental variable `AWS_SSM_MY_FIRST_PARAMTER` and `AWS_SSM_MY_SECOND_PARAMTER` with corresponding values from AWS SSM.
````
Watch for typo in `AWS_SSM_MY_FIRST_PARAMTER`

bomb-on commented on entrypoint.sh:
```diff
@@ -7,38 +7,47 @@ if [[ -z "$AWS_REGION" ]] || [[ -z "$AWS_ACCESS_KEY_ID" ]] || [[ -z "$AWS_SECRET
+}
+
+for parameter in $(echo $parameter_name_list | sed "s/,/ /g"); do
+get_ssm_param "$parameter"
```
Sorry for this, but can we just indent this line with 2 spaces?

bomb-on reviewed (changes requested):
Thanks again for updates, looks good to me apart from few changes I would do in README.md and that tiny indentation issue which is picking my eye ^^
Let me know what do you think about it too :)

craigtsmith commented:
Hey, apologies for taking so long to get around to this - I will work through the changes now.

craigtsmith committed:
Fix indentation

craigtsmith committed:
Tweak Readme, Fix typos.

craigtsmith committed:
Support new way of setting ENV variables

From 16/11/20 `::set-env` will be deprecated, see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

craigtsmith commented:
@bomb-on, I have added [this commit](https://github.com/bomb-on/aws-ssm-to-env/pull/3/commits/6a88fef9573b9add426a4561cf99cd29ff2dcd11), we were getting deprecation errors and noticed [this issue](https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/)

bomb-on commented:
Thanks for fixing all this, I wasn't aware of `set-env` alternative, I'm fine with it, just don't have time to test it out, did you maybe try to use it with new methods and does it work?

craigtsmith commented:
Yes it's tested and works as expected. They're removing the old version on Monday I think? So we had to fix it today.

bomb-on commented:
Ok, will approve it! Thank you one more time for you contributions and don't hesitate to propose additional functionality if you will need it. Unfortunately, I personally don't have a use case for this action anymore since I changed my job, but will keep my eye on future updates.

craigtsmith commented:
No problem! Thanks for creating the action! 
Góða helgi!

bomb-on commented:
hehe góða helgi sömuleiðis! :)
